src/config.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include <strings.h>
+#include "config.h"
+
+// Default configuration values.
+// All values provided by CONFIG PRD
+// (https://redislabs.atlassian.net/wiki/spaces/DX/pages/3980198010/PRD+CONFIG+for+modules+arguments)
+RM_Config rm_config = {
+    .bf_error_rate =
+        {
+            .value = 0.01,
+            .min = 0.0,
+            .max = 1.0,
+        },
+    .bf_initial_size =
+        {
+            .value = 100,
+            .min = 1,
+            .max = 1LL << 30,
+        },
+    .bf_expansion_factor =
+        {
+            .value = 2,
+            .min = 0,
+            .max = 32768,
+        },
+    .cf_bucket_size =
+        {
+            .value = 2,
+            .min = 1,
+            .max = 255,
+        },
+    .cf_initial_size =
+        {
+            .value = 1024,
+            .min = 4,
+            .max = 1LL << 30,
+        },
+    .cf_max_iterations =
+        {
+            .value = 20,
+            .min = 1,
+            .max = 65535,
+        },
+    .cf_expansion_factor =
+        {
+            .value = 1,
+            .min = 0,
+            .max = 32768,
+        },
+    .cf_max_expansions =
+        {
+            .value = 32,
+            .min = 1,
+            .max = 65535,
+        },
+};
+
+static int setFloatValue(const char *name, RedisModuleString *value, void *privdata,
+                         RedisModuleString **err) {
+    RM_ConfigFloat *config = privdata;
+    double new_val;
+    if (RedisModule_StringToDouble(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
```

Comment:
So the maximum and the minimum values are always inclusive, right? Then `cg_max_expansions` should have `.max = 65536` then, as well as others should be corrected? Do we have a test for the ranges?

Reply:
the max and min are always inclusive. thanks for noticing this one. there are no others that needed correcting. tests will be coming.